Remove wrong "disable collision" from right arm+hand SRDF

We should remove the last group of disable collisions, as they refer to links that don't exist in a biotac hand.

To avoid getting this:

```
[ERROR] [1433504838.775942782]: Link 'rh_mfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776000388]: Link 'rh_ffmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776027476]: Link 'rh_ffmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776063101]: Link 'rh_ffdistal' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776098811]: Link 'rh_rfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776129159]: Link 'rh_mfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776157383]: Link 'rh_mfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776190388]: Link 'rh_mfdistal' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776223425]: Link 'rh_lfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776253053]: Link 'rh_rfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776282747]: Link 'rh_rfmiddle' is not known to URDF. Cannot disable collisons.
[ERROR] [1433504838.776315468]: Link 'rh_rfdistal' is not known to URDF. Cannot disable collisons.
```

--- a/sr_description/hand/xacro/finger/biotac/biotac.urdf.xacro
+++ b/sr_description/hand/xacro/finger/biotac/biotac.urdf.xacro
@@ -14,7 +14,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
   <!-- parent is the parent link for the joint -->
   <xacro:macro name="biotac_middle"
   params="muscle prefix link_prefix joint_prefix parent">
-    <link name="${prefix}${link_prefix}adapter">
+    <link name="${prefix}${link_prefix}middle">
       <inertial>
         <origin xyz="0 0 0.0125" rpy="0 0 0" />
         <mass value="0.017" />
@@ -23,7 +23,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
       </inertial>
       <visual>
         <origin xyz="0 0 0" rpy="0 0 1.571" />
-        <geometry name="${prefix}${link_prefix}adapter_visual">
+        <geometry name="${prefix}${link_prefix}middle_visual">
           <mesh filename="package://sr_description/hand/model/biotacs/biotac_finger_adapter.dae"
           scale="1 1 1" />
         </geometry>
@@ -31,35 +31,35 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
       </visual>
       <collision>
         <origin xyz="0 0 0.005" rpy="0 0 0" />
-        <geometry name="${prefix}${link_prefix}adapter_collision_geom">
+        <geometry name="${prefix}${link_prefix}middle_collision_geom">
 
           <box size="0.001 0.001 0.001" />
         </geometry>
       </collision>
     </link>
-    <gazebo reference="${prefix}${link_prefix}adapter">
+    <gazebo reference="${prefix}${link_prefix}middle">
       <sensor type="contact"
-      name="${prefix}${link_prefix}adapter_contact">
+      name="${prefix}${link_prefix}middle_contact">
         <always_on>1</always_on>
         <update_rate>1000.0</update_rate>
         <contact>
           <collision>
-          ${prefix}${link_prefix}adapter_collision</collision>
-          <topic>${prefix}${link_prefix}adapter_collision</topic>
+          ${prefix}${link_prefix}middle_collision</collision>
+          <topic>${prefix}${link_prefix}middle_collision</topic>
         </contact>
-        <plugin name="${prefix}${link_prefix}adapter_gazebo_ros_bumper_controller"
+        <plugin name="${prefix}${link_prefix}middle_gazebo_ros_bumper_controller"
         filename="libgazebo_ros_bumper.so">
           <updateRate>100.0</updateRate>
-          <frameName>${prefix}${link_prefix}adapter</frameName>
+          <frameName>${prefix}${link_prefix}middle</frameName>
           <bumperTopicName>
-          contacts/${prefix}${link_prefix}/adapter</bumperTopicName>
+          contacts/${prefix}${link_prefix}/middle</bumperTopicName>
         </plugin>
       </sensor>
       <material>Gazebo/FlatBlack</material>
     </gazebo>
     <joint name="${prefix}${joint_prefix}J2" type="revolute">
       <parent link="${parent}" />
-      <child link="${prefix}${link_prefix}adapter" />
+      <child link="${prefix}${link_prefix}middle" />
       <origin xyz="0 0 0.045" rpy="0 0 0" />
       <axis xyz="1 0 0" />
       <limit lower="0" upper="${M_PI/2}" effort="2.0"
@@ -79,7 +79,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
       </inertial>
     </link>
     <joint name="${prefix}${joint_prefix}J1" type="revolute">
-      <parent link="${prefix}${link_prefix}adapter" />
+      <parent link="${prefix}${link_prefix}middle" />
       <child link="${prefix}${link_prefix}_J1_dummy" />
       <origin xyz="0 0 0.02" rpy="0 0 0" />
       <axis xyz="1 0 0" />
@@ -94,7 +94,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
     <xacro:middle_transmission muscletrans="${muscle}"
     prefix="${prefix}" joint_prefix="${joint_prefix}"
     link_prefix="${link_prefix}" />
-    <link name="${prefix}${link_prefix}biotac">
+    <link name="${prefix}${link_prefix}distal">
       <inertial>
         <origin xyz="0.0 -0.009 0.035" rpy="0 0 0" />
         <!-- approximated only FIXME -->
@@ -104,14 +104,14 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
       </inertial>
       <visual>
         <origin xyz="0 0 0" rpy="0 0 0" />
-        <geometry name="${prefix}${link_prefix}biotac_visual">
+        <geometry name="${prefix}${link_prefix}distal_visual">
           <!-- <mesh filename="package://sr_description/hand/model/biotacs/biotac.stl" scale="0.0254 0.0254 0.0254" /> -->
           <mesh filename="package://sr_description/hand/model/biotacs/biotac_decimated.dae"
           scale="0.0254 0.0254 0.0254" />
         </geometry>
       </visual>
       <collision>
-        <geometry name="${prefix}${link_prefix}biotac_collision_geom">
+        <geometry name="${prefix}${link_prefix}distal_collision_geom">
 
           <!-- <mesh filename="package://sr_description/hand/model/biotacs/biotac.stl" scale="0.0254 0.0254 0.0254" /> -->
           <mesh filename="package://sr_description/hand/model/biotacs/biotac_decimated.dae"
@@ -119,30 +119,30 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
         </geometry>
       </collision>
     </link>
-    <gazebo reference="${prefix}${link_prefix}biotac">
+    <gazebo reference="${prefix}${link_prefix}distal">
       <sensor type="contact"
-      name="${prefix}${link_prefix}biotac_contact">
+      name="${prefix}${link_prefix}distal_contact">
         <always_on>1</always_on>
         <update_rate>1000.0</update_rate>
         <contact>
           <collision>
-          ${prefix}${link_prefix}biotac_collision</collision>
-          <topic>${prefix}${link_prefix}biotac_collision</topic>
+          ${prefix}${link_prefix}distal_collision</collision>
+          <topic>${prefix}${link_prefix}distal_collision</topic>
         </contact>
-        <plugin name="${prefix}${link_prefix}biotac_gazebo_ros_bumper_controller"
+        <plugin name="${prefix}${link_prefix}distal_gazebo_ros_bumper_controller"
         filename="libgazebo_ros_bumper.so">
           <updateRate>100.0</updateRate>
-          <frameName>${prefix}${link_prefix}biotac</frameName>
+          <frameName>${prefix}${link_prefix}distal</frameName>
           <bumperTopicName>
-          contacts/${prefix}${link_prefix}/biotac</bumperTopicName>
+          contacts/${prefix}${link_prefix}/distal</bumperTopicName>
         </plugin>
       </sensor>
       <selfCollide>true</selfCollide>
     </gazebo>
-    <joint name="${prefix}${joint_prefix}_biotac_joint"
+    <joint name="${prefix}${joint_prefix}_distal_joint"
     type="fixed">
-      <parent link="${prefix}${link_prefix}adapter" />
-      <child link="${prefix}${link_prefix}biotac" />
+      <parent link="${prefix}${link_prefix}middle" />
+      <child link="${prefix}${link_prefix}distal" />
       <origin xyz="0 0.0 0.008" rpy="0 0 0" />
     </joint>
     <!-- extra link to imaginary sphere in fingertip for FK/IK calculations -->
@@ -165,7 +165,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
     </gazebo>
     <joint name="${prefix}${link_prefix}tip_fixed_joint"
     type="fixed">
-      <parent link="${prefix}${link_prefix}adapter" />
+      <parent link="${prefix}${link_prefix}middle" />
       <child link="${prefix}${link_prefix}tip" />
       <origin xyz="0.0 -0.009 ${0.008+0.044}" rpy="0 0 -1.571" />
       <!-- approximated only FIXME -->

--- a/sr_description/hand/xacro/finger/biotac/biotac.urdf.xacro
+++ b/sr_description/hand/xacro/finger/biotac/biotac.urdf.xacro
@@ -160,10 +160,10 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
         </geometry>
       </visual>
     </link>
-    <gazebo reference="${prefix}${link_prefix}tip">
+    <gazebo reference="${prefix}${joint_prefix}tip">
       <material name="Gazebo/RedGlow" />
     </gazebo>
-    <joint name="${prefix}${link_prefix}tip_fixed_joint"
+    <joint name="${prefix}${joint_prefix}tip"
     type="fixed">
       <parent link="${prefix}${link_prefix}middle" />
       <child link="${prefix}${link_prefix}tip" />

--- a/sr_description/hand/xacro/thumb/thdistal.urdf.xacro
+++ b/sr_description/hand/xacro/thumb/thdistal.urdf.xacro
@@ -19,7 +19,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
   params="muscle bio bt_sp ubi eli prefix parent">
     <xacro:if value="${bio}">
       <!--    BIOTAC -->
-      <link name="${prefix}thbiotac_J1_dummy">
+      <link name="${prefix}thdistal_J1_dummy">
         <inertial>
           <origin xyz="0 0 0" rpy="0 0 0" />
           <mass value="0.001" />
@@ -29,7 +29,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
       </link>
       <joint name="${prefix}THJ1" type="revolute">
         <parent link="${parent}" />
-        <child link="${prefix}thbiotac_J1_dummy" />
+        <child link="${prefix}thdistal_J1_dummy" />
         <origin xyz="0 0 0.02" rpy="0 0 ${-M_PI/2}" />
         <axis xyz="1 0 0" />
         <limit lower="${20*M_PI/180}" upper="${20.1*M_PI/180}"
@@ -40,7 +40,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
         <provideFeedback>1</provideFeedback>
         <implicitSpringDamper>1</implicitSpringDamper>
       </gazebo>
-      <link name="${prefix}thbiotac">
+      <link name="${prefix}thdistal">
         <inertial>
           <origin xyz="-0.009 0.0 0.035" rpy="0 0 ${-M_PI/2}" />
           <!-- approximated only FIXME -->
@@ -50,7 +50,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
         </inertial>
         <visual>
           <origin xyz="0 0 0" rpy="0 0 0" />
-          <geometry name="${prefix}th_biotac_visual">
+          <geometry name="${prefix}th_distal_visual">
             <!-- <mesh filename="package://sr_description/hand/model/biotacs/biotac.stl" scale="0.0254 0.0254 0.0254" /> -->
             <mesh filename="package://sr_description/hand/model/biotacs/biotac_decimated.dae"
             scale="0.0254 0.0254 0.0254" />
@@ -58,7 +58,7 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
           <material name="BiotacGreen" />
         </visual>
         <collision>
-          <geometry name="${prefix}th_biotac_collision_geom">
+          <geometry name="${prefix}th_distal_collision_geom">
             <!-- <box size="0.001 0.001 0.001" /> -->
             <!-- <mesh filename="package://sr_description/hand/model/biotacs/biotac-v5.stl" scale="0.0254 0.0254 0.0254" /> -->
             <mesh filename="package://sr_description/hand/model/biotacs/biotac_decimated.dae"
@@ -71,13 +71,13 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
           <always_on>1</always_on>
           <update_rate>1000.0</update_rate>
           <contact>
-            <collision>${prefix}thbiotac_geom</collision>
-            <topic>${prefix}thbiotac_collision</topic>
+            <collision>${prefix}thdistal_geom</collision>
+            <topic>${prefix}thdistal_collision</topic>
           </contact>
-          <plugin name="${prefix}thbiotac_gazebo_ros_bumper_controller"
+          <plugin name="${prefix}thdistal_gazebo_ros_bumper_controller"
           filename="libgazebo_ros_bumper.so">
             <alwaysOn>true</alwaysOn>
-            <frameName>${prefix}thbiotac</frameName>
+            <frameName>${prefix}thdistal</frameName>
             <updateRate>100.0</updateRate>
             <bumperTopicName>
             contacts/${prefix}th/distal</bumperTopicName>
@@ -86,9 +86,9 @@ xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom">
         <selfCollide>true</selfCollide>
         <material>BiotacGreen</material>
       </gazebo>
-      <joint name="${prefix}th_biotac_joint" type="fixed">
+      <joint name="${prefix}th_distal_joint" type="fixed">
         <parent link="${prefix}thmiddle" />
-        <child link="${prefix}thbiotac" />
+        <child link="${prefix}thdistal" />
         <origin xyz="0 0.0 0.01" rpy="0 0 ${-M_PI/2}" />
       </joint>
       <!-- extra link to imaginary sphere in fingertip for FK/IK calculations -->

--- a/sr_multi_moveit/right_sr_ur10_moveit_config/config/ur10srh.srdf
+++ b/sr_multi_moveit/right_sr_ur10_moveit_config/config/ur10srh.srdf
@@ -26,19 +26,23 @@
         <joint name="rh_FFJ3" />
         <joint name="rh_FFJ2" />
         <joint name="rh_FFJ1" />
+        <joint name="rh_FFtip" />
         <joint name="rh_LFJ5" />
         <joint name="rh_LFJ4" />
         <joint name="rh_LFJ3" />
         <joint name="rh_LFJ2" />
         <joint name="rh_LFJ1" />
+        <joint name="rh_LFtip" />
         <joint name="rh_MFJ4" />
         <joint name="rh_MFJ3" />
         <joint name="rh_MFJ2" />
         <joint name="rh_MFJ1" />
+        <joint name="rh_MFtip" />
         <joint name="rh_RFJ4" />
         <joint name="rh_RFJ3" />
         <joint name="rh_RFJ2" />
         <joint name="rh_RFJ1" />
+        <joint name="rh_RFtip" />
         <joint name="rh_THJ5" />
         <joint name="rh_THJ4" />
         <joint name="rh_THJ3" />
@@ -46,7 +50,8 @@
         <joint name="rh_THJ1" />
         <joint name="rh_thtip" />
     </group>
-    <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
+
+        <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
     <group_state name="open" group="right_hand">
         <joint name="rh_FFJ1" value="0" />
         <joint name="rh_FFJ2" value="0" />
@@ -225,6 +230,7 @@
         <joint name="ra_wrist_2_joint" value="0" />
         <joint name="ra_wrist_3_joint" value="0" />
     </group_state>
+
     <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
     <end_effector name="right_hand" parent_link="rh_forearm" group="right_hand" />
     <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
@@ -238,405 +244,255 @@
     <disable_collisions link1="ra_wrist_2_link" link2="ra_wrist_3_link" reason="Adjacent" />
     <disable_collisions link1="ra_wrist_2_link" link2="rh_forearm" reason="Default" />
     <disable_collisions link1="ra_wrist_3_link" link2="rh_forearm" reason="Adjacent" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_ffbiotac" reason="Adjacent" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_ffknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_ffproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_fftip" reason="Adjacent" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_forearm" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_lfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_lfmetacarpal" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_rftip" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_ffadapter" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_ffknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_ffproximal" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_fftip" reason="Default" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_forearm" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lfadapter" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lfmetacarpal" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_lftip" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_mfadapter" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_mfproximal" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_mftip" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_rfadapter" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_rftip" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thhub" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thmiddle" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thproximal" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_thtip" reason="User" />
-    <disable_collisions link1="rh_ffbiotac" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_ffknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_ffmiddle" reason="Adjacent" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_ffproximal" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_forearm" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_lfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_lfproximal" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_rfproximal" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_ffdistal" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_ffmiddle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_ffproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_fftip" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_forearm" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_lfadapter" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_lfbiotac" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_lfdistal" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_lfknuckle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_lfmiddle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_mfadapter" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_mfbiotac" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_mfdistal" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_mfmiddle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_mftip" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_palm" reason="Adjacent" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_rfbiotac" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_rfdistal" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffknuckle" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_thbiotac" reason="User" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_ffknuckle" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_ffknuckle" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_fftip" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_ffproximal" reason="Adjacent" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_forearm" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_lfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_lfproximal" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_rfproximal" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_ffmiddle" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_forearm" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_lfadapter" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_lfbiotac" reason="Never" />
+    <disable_collisions link1="rh_ffproximal" link2="rh_lfdistal" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_lfknuckle" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_ffproximal" link2="rh_lfmiddle" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_mftip" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_palm" reason="User" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_rfbiotac" reason="Never" />
+    <disable_collisions link1="rh_ffproximal" link2="rh_rfdistal" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_ffproximal" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_thbiotac" reason="User" />
     <disable_collisions link1="rh_ffproximal" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_ffproximal" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_thproximal" reason="Never" />
     <disable_collisions link1="rh_ffproximal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_forearm" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_lfknuckle" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_lfmetacarpal" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_fftip" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_fftip" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_fftip" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_fftip" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_lfadapter" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_lfbiotac" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_lfdistal" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_lfknuckle" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_lfmiddle" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_mfadapter" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_mfbiotac" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_mfdistal" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_mfmiddle" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_mftip" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_rfbiotac" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_rfdistal" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_thbiotac" reason="Never" />
+    <disable_collisions link1="rh_forearm" link2="rh_thdistal" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_forearm" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_forearm" link2="rh_wrist" reason="Adjacent" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_lfbiotac" reason="Adjacent" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_lfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_lfmetacarpal" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_lfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_lftip" reason="Adjacent" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_lfadapter" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_lfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_lfmetacarpal" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_lfproximal" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_lftip" reason="Default" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_mfadapter" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_mfproximal" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_mftip" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_palm" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_rfadapter" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_rfproximal" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_rftip" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_thtip" reason="User" />
-    <disable_collisions link1="rh_lfbiotac" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_lfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_lfmetacarpal" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_lfmiddle" reason="Adjacent" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_lfproximal" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_mfproximal" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_thmiddle" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_lfmetacarpal" reason="Adjacent" />
+    <disable_collisions link1="rh_lfknuckle" link2="rh_lfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_lfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_mfadapter" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_mfbiotac" reason="Never" />
+    <disable_collisions link1="rh_lfknuckle" link2="rh_mfdistal" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfknuckle" link2="rh_mfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_mftip" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_palm" reason="User" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_rfbiotac" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_rftip" reason="Never" />
+    <disable_collisions link1="rh_lfknuckle" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_thbiotac" reason="Never" />
+    <disable_collisions link1="rh_lfknuckle" link2="rh_thdistal" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_lfknuckle" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_lfknuckle" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_lfmetacarpal" link2="rh_lfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_lfproximal" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfadapter" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfbiotac" reason="Never" />
+    <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfdistal" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_mftip" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_palm" reason="Adjacent" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_rfbiotac" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfmetacarpal" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_thbiotac" reason="Never" />
+    <disable_collisions link1="rh_lfmetacarpal" link2="rh_thdistal" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_lfmetacarpal" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_lfmetacarpal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_lfproximal" link2="rh_lftip" reason="Never" />
-    <disable_collisions link1="rh_lfproximal" link2="rh_mfbiotac" reason="User" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_lfproximal" reason="Adjacent" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_thmiddle" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_lfmiddle" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_lfproximal" link2="rh_mfknuckle" reason="Never" />
     <disable_collisions link1="rh_lfproximal" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_lfproximal" link2="rh_rfbiotac" reason="User" />
     <disable_collisions link1="rh_lfproximal" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lfproximal" link2="rh_thbiotac" reason="User" />
     <disable_collisions link1="rh_lfproximal" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_lfproximal" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_lfproximal" link2="rh_thproximal" reason="Never" />
     <disable_collisions link1="rh_lfproximal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_mfbiotac" reason="User" />
-    <disable_collisions link1="rh_lftip" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_mfproximal" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_palm" reason="User" />
-    <disable_collisions link1="rh_lftip" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_lftip" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_lftip" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_lftip" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_mfbiotac" reason="Adjacent" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_mfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_mftip" reason="Adjacent" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_mfadapter" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_mfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_mfproximal" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_mftip" reason="Default" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_rfadapter" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_rfproximal" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_rftip" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thhub" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thmiddle" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_thtip" reason="User" />
-    <disable_collisions link1="rh_mfbiotac" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_mfknuckle" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_mfmiddle" reason="Adjacent" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_mfproximal" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_mfknuckle" link2="rh_mfmiddle" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_mfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_mftip" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_palm" reason="Adjacent" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_rfbiotac" reason="Never" />
+    <disable_collisions link1="rh_mfknuckle" link2="rh_rfdistal" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_rftip" reason="Never" />
+    <disable_collisions link1="rh_mfknuckle" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_thbiotac" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_mfknuckle" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_mfknuckle" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_mftip" reason="Never" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_palm" reason="User" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_rfbiotac" reason="User" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_mfproximal" reason="Adjacent" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_palm" reason="Never" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_mfmiddle" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_mfproximal" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_mfproximal" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_thbiotac" reason="User" />
     <disable_collisions link1="rh_mfproximal" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_mfproximal" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_mfproximal" link2="rh_thproximal" reason="Never" />
     <disable_collisions link1="rh_mfproximal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_palm" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_rfbiotac" reason="User" />
-    <disable_collisions link1="rh_mftip" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_rfproximal" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_mftip" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_mftip" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_palm" link2="rh_rfadapter" reason="Never" />
-    <disable_collisions link1="rh_palm" link2="rh_rfbiotac" reason="Never" />
+    <disable_collisions link1="rh_palm" link2="rh_rfdistal" reason="Never" />
     <disable_collisions link1="rh_palm" link2="rh_rfknuckle" reason="Adjacent" />
-    <disable_collisions link1="rh_palm" link2="rh_rfproximal" reason="User" />
-    <disable_collisions link1="rh_palm" link2="rh_rftip" reason="Never" />
+    <disable_collisions link1="rh_palm" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_palm" link2="rh_thbase" reason="Adjacent" />
-    <disable_collisions link1="rh_palm" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_palm" link2="rh_thhub" reason="User" />
-    <disable_collisions link1="rh_palm" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_palm" link2="rh_thproximal" reason="User" />
-    <disable_collisions link1="rh_palm" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_palm" link2="rh_wrist" reason="Adjacent" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_rfbiotac" reason="Adjacent" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_rfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_rftip" reason="Adjacent" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_rfadapter" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_rfknuckle" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_rfproximal" reason="User" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_rftip" reason="Default" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thbiotac" reason="User" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_thtip" reason="User" />
-    <disable_collisions link1="rh_rfbiotac" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_rfknuckle" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_rfmiddle" reason="Adjacent" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_rfproximal" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_thmiddle" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_rfknuckle" link2="rh_rfmiddle" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_rfproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_rfknuckle" link2="rh_rftip" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_rfknuckle" link2="rh_thbiotac" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_rfknuckle" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_rfknuckle" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_rfproximal" link2="rh_rftip" reason="Never" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_rfproximal" reason="Adjacent" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_thbase" reason="Never" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_thmiddle" reason="Never" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_rfmiddle" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_rfproximal" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_rfproximal" link2="rh_thbiotac" reason="User" />
     <disable_collisions link1="rh_rfproximal" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_rfproximal" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_rfproximal" link2="rh_thproximal" reason="Never" />
     <disable_collisions link1="rh_rfproximal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thbase" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thbiotac" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thmiddle" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_thtip" reason="Never" />
-    <disable_collisions link1="rh_rftip" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_thbase" link2="rh_thbiotac" reason="Never" />
+    <disable_collisions link1="rh_thbase" link2="rh_thdistal" reason="Never" />
     <disable_collisions link1="rh_thbase" link2="rh_thhub" reason="Never" />
     <disable_collisions link1="rh_thbase" link2="rh_thmiddle" reason="Never" />
     <disable_collisions link1="rh_thbase" link2="rh_thproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_thbase" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_thbase" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_thbiotac" link2="rh_thhub" reason="Never" />
-    <disable_collisions link1="rh_thbiotac" link2="rh_thmiddle" reason="Adjacent" />
-    <disable_collisions link1="rh_thbiotac" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_thbiotac" link2="rh_thtip" reason="Default" />
-    <disable_collisions link1="rh_thbiotac" link2="rh_wrist" reason="Never" />
+    <disable_collisions link1="rh_thdistal" link2="rh_thhub" reason="Never" />
+    <disable_collisions link1="rh_thdistal" link2="rh_thmiddle" reason="Adjacent" />
+    <disable_collisions link1="rh_thdistal" link2="rh_thproximal" reason="Never" />
+    <disable_collisions link1="rh_thdistal" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_thhub" link2="rh_thmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_thhub" link2="rh_thproximal" reason="Adjacent" />
-    <disable_collisions link1="rh_thhub" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_thhub" link2="rh_wrist" reason="Never" />
     <disable_collisions link1="rh_thmiddle" link2="rh_thproximal" reason="Never" />
-    <disable_collisions link1="rh_thmiddle" link2="rh_thtip" reason="Adjacent" />
     <disable_collisions link1="rh_thmiddle" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_thproximal" link2="rh_thtip" reason="Never" />
     <disable_collisions link1="rh_thproximal" link2="rh_wrist" reason="Never" />
-    <disable_collisions link1="rh_thtip" link2="rh_wrist" reason="Never" />
 
     <disable_collisions link1="rh_ffproximal" link2="rh_mfproximal" reason="Adjacent" />
     <disable_collisions link1="rh_ffproximal" link2="rh_mfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_ffmiddle" link2="rh_mfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_ffmiddle" link2="rh_mfdistal" reason="Adjacent" />
     <disable_collisions link1="rh_ffdistal" link2="rh_mfdistal" reason="Adjacent" />
-    
+
     <disable_collisions link1="rh_mfproximal" link2="rh_rfproximal" reason="Adjacent" />
     <disable_collisions link1="rh_mfproximal" link2="rh_rfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_mfmiddle" link2="rh_rfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_mfmiddle" link2="rh_rfdistal" reason="Adjacent" />
     <disable_collisions link1="rh_mfdistal" link2="rh_rfdistal" reason="Adjacent" />
-    
+
     <disable_collisions link1="rh_rfproximal" link2="rh_lfproximal" reason="Adjacent" />
     <disable_collisions link1="rh_rfproximal" link2="rh_lfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_rfmiddle" link2="rh_lfmiddle" reason="Adjacent" />
     <disable_collisions link1="rh_rfmiddle" link2="rh_lfdistal" reason="Adjacent" />
     <disable_collisions link1="rh_rfdistal" link2="rh_lfdistal" reason="Adjacent" />
-    
+
+
     <disable_collisions link1="rh_ffproximal" link2="rh_palm" reason="Adjacent" />
-    <disable_collisions link1="rh_mfproximal" link2="rh_palm" reason="Adjacent" />    
+    <disable_collisions link1="rh_mfproximal" link2="rh_palm" reason="Adjacent" />
     <disable_collisions link1="rh_rfproximal" link2="rh_palm" reason="Adjacent" />
-    <disable_collisions link1="rh_lfproximal" link2="rh_palm" reason="Adjacent" />   
-    <disable_collisions link1="rh_thproximal" link2="rh_palm" reason="Adjacent" />   
-      
+    <disable_collisions link1="rh_lfproximal" link2="rh_palm" reason="Adjacent" />
+    <disable_collisions link1="rh_thproximal" link2="rh_palm" reason="Adjacent" />
+
+    <disable_collisions link1="rh_ffdistal" link2="rh_fftip" reason="Adjacent" />
+    <disable_collisions link1="rh_mfdistal" link2="rh_mftip" reason="Adjacent" />
+    <disable_collisions link1="rh_rfdistal" link2="rh_rftip" reason="Adjacent" />
+    <disable_collisions link1="rh_lfdistal" link2="rh_lftip" reason="Adjacent" />
+    <disable_collisions link1="rh_thdistal" link2="rh_thtip" reason="Adjacent" />
 </robot>

